game-core/src/test/java/games/strategy/triplea/delegate/battle/MustFightBattleExecutablesTest.java

```diff
@@ -0,0 +1,1670 @@
+  }
+
+  @Test
+  @DisplayName(
+      "When attacker has a destroyer, defender has a destroyer, WW2v2 is true, "
+          + "and DEFENDING_SUBS_SNEAK_ATTACK is either, then attacker>defender>standard")
+  void firstStrikeOrderAttHasDestroyerDefHasDestroyerWW2v2TrueSneakAttackTrueFalse() {
+    assertThatFirstStrikeStepOrder(
+        givenFirstStrikeBattleSetup(true, true, true, true, true),
+        List.of(
+            FirstStrikeBattleStep.ATTACKER,
+            FirstStrikeBattleStep.DEFENDER,
+            FirstStrikeBattleStep.STANDARD));
+  }
+
+  private Tuple<MustFightBattle, List<IExecutable>> givenFirstStrikeBattleSetup(
+      final boolean attackerDestroyer,
+      final boolean defenderDestroyer,
+      final boolean ww2v2,
+      final boolean defendingSubsSneakAttack,
+      final boolean ignoreDefendingSubsSneakAttack) {
+    final MustFightBattle battle = spy(newBattle(WATER));
+    lenient().doNothing().when(battle).firstStrikeAttackersFire(any());
+    lenient().doNothing().when(battle).firstStrikeDefendersFire(any());
+
+    when(gameProperties.get(SUB_RETREAT_BEFORE_BATTLE, false)).thenReturn(false);
+    when(gameProperties.get(TRANSPORT_CASUALTIES_RESTRICTED, false)).thenReturn(false);
+    when(gameProperties.get(WW2V2, false)).thenReturn(ww2v2);
+    if (!ignoreDefendingSubsSneakAttack) {
+      when(gameProperties.get(DEFENDING_SUBS_SNEAK_ATTACK, false))
+          .thenReturn(defendingSubsSneakAttack);
+    }
+
+    final Unit attackerUnit = attackerDestroyer ? givenUnitDestroyer() : givenUnit();
+    final Unit defenderUnit = defenderDestroyer ? givenUnitDestroyer() : givenUnit();
+
+    battle.setUnits(
+        List.of(defenderUnit), List.of(attackerUnit), List.of(), List.of(), defender, List.of());
+    final List<IExecutable> execs = battle.getBattleExecutables(true);
+
+    return Tuple.of(battle, execs);
+  }
+
+  private enum FirstStrikeBattleStep {
+    ATTACKER,
+    DEFENDER,
+    STANDARD,
+  }
+
+  private void assertThatFirstStrikeStepOrder(
+      final Tuple<MustFightBattle, List<IExecutable>> battleTuple,
+      final List<FirstStrikeBattleStep> stepOrder) {
+    final List<IExecutable> execs = battleTuple.getSecond();
+
+    final EnumMap<FirstStrikeBattleStep, Integer> indices =
+        new EnumMap<>(FirstStrikeBattleStep.class);
+
+    indices.put(
+        FirstStrikeBattleStep.ATTACKER,
+        getIndex(execs, MustFightBattle.FirstStrikeAttackersFire.class));
+    indices.put(
+        FirstStrikeBattleStep.DEFENDER,
+        getIndex(execs, MustFightBattle.FirstStrikeDefendersFire.class));
+    indices.put(
+        FirstStrikeBattleStep.STANDARD,
+        getIndex(execs, MustFightBattle.StandardAttackersFire.class));
+
+    assertThat(indices.get(stepOrder.get(0)), lessThan(indices.get(stepOrder.get(1))));
+    assertThat(indices.get(stepOrder.get(1)), lessThan(indices.get(stepOrder.get(2))));
+  }
+
+  private void assertThatFirstStrikeReturnFireIs(
```

Comment:
A few nits here:

1. The boolean parameter is a bit hard to grok. Specifically "is attacker = true" is not necessarily clear until you look at the implementation in detail.

2. We are not really asserting anything in this method. I initially wondered if a matcher could replace this method, but instead we are doing a mockito verify.

What do you think of renaming to "verify" instead of "assert" as the method prefix?

I'd also suggest the boolean parameter be removed simply by creating additional methods that can be called directly to then call the underlying method, ie:
```
  private void verifyAttackersFireWithFirstStrike(
      final Tuple<MustFightBattle, List<IExecutable>> battleTuple,
      final MustFightBattle.ReturnFire returnFire) {

    verifyThatFirstStrikeReturnFireIs(battleTuple, returnFire, true);
  }
```
WDYT? Is there a cleaner way to get rid of the boolean parameter from the test cases?